Hyper/MarchingCubes/ChunkWorker.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Reflection.Metadata;
+using System.Threading;
+using Hyper.Collisions;
+using Hyper.MarchingCubes;
+using Hyper.Meshes;
+using OpenTK.Mathematics;
+using Hyper.Collisions.Bepu;
+
+namespace Hyper
+{
+    internal class ChunkWorker
```

Comment:
Have you considered using the [producer consumer pattern](https://jenkov.com/tutorials/java-concurrency/producer-consumer.html) here? It seems like a good place for it.
The producer thread would determine chunk visibility based on distance and send things like (remove, (0,1,0)), (add, (2, 0, 0)) to the message queue. Then the consumer would perform chunk addition/removal. I found implementation ideas here https://stackoverflow.com/questions/1656404/c-sharp-producer-consumer but it seems that there's something like that already provided by C# : https://learn.microsoft.com/en-us/dotnet/standard/collections/thread-safe/blockingcollection-overview?redirectedfrom=MSDN.

Reply:
you are totally right